lib/internal/Magento/Framework/ObjectManager/InterceptableValidator.php

```diff
@@ -28,6 +27,20 @@ private function isInterceptor($instanceName)
         return $this->endsWith($instanceName, '\Interceptor');
     }
 
+    /**
+     * Check if instance type is proxy
+     *
+     * @param string $instanceName
+     * @return bool
+     */
+    private function isProxy($instanceName)
+    {
+        return $this->endsWith(
+            $instanceName,
+            '\\' . ucfirst(\Magento\Framework\ObjectManager\Code\Generator\Proxy::ENTITY_TYPE)
+        );
+    }
```

Comment:
I believe performance should be really good here, so it's better to not calculate all the time ucfirst.
```suggestion
        return $this->endsWith($instanceName, '\Proxy');
    }
```

Reply:
Ok, I'm changing this.